packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte

```diff
@@ -0,0 +1,302 @@
+<script lang="ts">
+import {
+  faArrowUpRightFromSquare,
+  faEllipsisVertical,
+  faPlay,
+  faRotateRight,
+  faStop,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons';
+import Fa from 'svelte-fa/src/fa.svelte';
+import { providerInfos } from '../../stores/providers';
+import type { ProviderContainerConnectionInfo, ProviderInfo } from '../../../../main/src/plugin/api/provider-info';
+import { onMount } from 'svelte';
+import type {
+  IConfigurationPropertyRecordedSchema,
+  IProviderContainerConfigurationPropertyRecordedSchema,
+} from '../../../../main/src/plugin/configuration-registry';
+import { configurationProperties } from '../../stores/configurationProperties';
+import type { ContainerProviderConnection } from '@tmpwip/extension-api';
+
+interface containerStatus {
+  status: string;
+  inProgress: boolean;
+}
+
+export let properties: IConfigurationPropertyRecordedSchema[] = [];
+let providers: ProviderInfo[] = [];
+$: containerConnectionStatus = new Map<string, containerStatus>();
+
+let configurationKeys: IConfigurationPropertyRecordedSchema[];
+$: containerConnectionDivWidth = 0;
+$: containerConnectionDivHeight = 0;
+
+onMount(() => {
+  configurationProperties.subscribe(value => {
+    properties = value;
+  });
+
+  providerInfos.subscribe(providerInfosValue => {
+    providers = providerInfosValue;
+    providers.forEach(provider => {
+      provider.containerConnections.forEach(container => {
+        // update the map only if the container state is different from last time
+        if (
+          containerConnectionStatus.has(`${provider.name}-${container.name}`) &&
+          containerConnectionStatus.get(`${provider.name}-${container.name}`).status !== container.status
+        ) {
+          containerConnectionStatus.set(`${provider.name}-${container.name}`, {
+            inProgress: false,
+            status: container.status,
+          });
+        }
+      });
+    });
+    containerConnectionStatus = containerConnectionStatus;
+  });
+});
+
+$: configurationKeys = properties
+  .filter(property => property.scope === 'ContainerConnection')
+  .sort((a, b) => a.id.localeCompare(b.id));
+
+let tmpProviderContainerConfiguration: IProviderContainerConfigurationPropertyRecordedSchema[] = [];
```

Comment:
looks like the interface you declared globally `IProviderContainerConfigurationPropertyRecordedSchema` is only used there in the UI so probably you need to declare it only here